domain-knowledge/DOMAIN-OBJECTS.md

```diff
@@ -0,0 +1,501 @@
+- `Profile` `belongsTo()` `Country`
+
+#### Access Policy
+
+- Any `Artist` or `Label` may create a `Profile`.
+- Anybody may view a `Profile`.
+- An `Artist` or `Label` may update its own `Profile`.
+- An `Artist` or `Label` may force-delete its own `Profile`.
+- When an `Artist` or `Label` is deleted, any associated `Profile` is likewise
+  deleted.
+
+### Song
+
+A `Song` is exactly what it sounds like. An `Album` contains one or more `Songs`.
+
+An `Artist` may disable a `Song` at any time, which removes it from public
+visibility and prevents it from being added to a customer's shopping cart.
+Disabling a `Song` has no effect for customers who have already purchased the
+`Song`, and it will continue to appear in their "Purchased Music" history.
+An `Artist` may want to disable a `Song` for any number of reasons,
+ranging from simply not wanting it to be available anymore, to addressing a
+copyright infringement claim that does not affect any other track on the `Album`
+(in which case disabling or deleting the entire `Album` may not be necessary).
+
+Each `Song` has its own price, which the owning `Artist` specifies.
+The minimum `Song` price is $0.35, which is necessary to ensure that the payment
+processing cost for a `Song` purchased individually does not exceed its price.
+In other words, if a `Song` is priced at $0.05, for example, but the minimum payment
+processing charge is 2.9% + $0.30 (this is a typical real-world rate),
+payment processing would cost $0.31, thereby yielding a $0.26 loss for the service
+operator and zero profit for the `Artist`.
+
+The astute reader may be wondering why individual `Songs` cannot be priced
+under $0.35 as long as they are purchased in quantities sufficient to offset the
+payment processing cost. Technically, this is possible to implement, but in practice,
+it leads to a confusing shopping experience because there can be no up-front
+price listed for `Songs` under $0.35 (given that the discount can be calculated
+only after `Songs` are added to the shopping cart). Even if customers are shown
+the $0.35 price prior to adding the `Song` to the cart, and any applicable
+discount is applied thereafter, the reason for the discount is confusing and
+unintuitive. Accordingly, the authors have chosen not to pursue such a pricing
+structure as yet.
+
+#### Relationships
+
+- `Song` `belongsTo()` `Album`
+- `Song` `belongsTo()` `FlacFile`
+- `Song` `morphMany()` copies sold (`DigitalAsset`) 
+- `Song` `morphOne()` asset (`DigitalAsset`) 
+
+#### Access Policy
+
+- Any `Artist` may create a `Song`.
+- Anybody may view a `Song` if its `is_active` flag is truthy, but only as long
+  as the related `Album`'s `is_active` flag is also truthy.
+- If a `Song`'s or its related `Album`'s `is_active` flag is falsey, only the
+  owning `Artist`, or anybody who has purchased access to it, may view the `Song`.
+- All rules that apply to the `is_active` flag apply equally to the
+  soft-deletion (`deleted_at`) flag.
```

Comment:
Suggested change:

"...flag apply equally but inversely to the soft-deletion (`deleted_at`) flag."

(_equally but_, can be left out)

Otherwise a truthy deleted_at === can be viewed.

Reply:
You're almost never wrong, so let me take a closer look at this... :)

Reply:
Maybe this is the first time. So please do check :D